supersede.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/taskcluster/generic-worker/gwconfig"
+	"github.com/taskcluster/httpbackoff"
+	"github.com/taskcluster/taskcluster-base-go/scopes"
+)
+
+var (
+	supersededByPath = filepath.Join("generic-worker", "superseded-by.json")
+	supersededByName = "public/superseded-by.json"
+)
+
+type SupersedeFeature struct {
+}
+
+type SupersedesServiceResponse struct {
+	TaskIDs []string `json:"supersedes"`
+}
+
+func (feature *SupersedeFeature) Name() string {
+	return "Supersede"
+}
+
+func (feature *SupersedeFeature) Initialise() error {
+	return nil
+}
+
+func (feature *SupersedeFeature) PersistState() error {
+	return nil
+}
+
+// Supersede is always enabled
+func (feature *SupersedeFeature) IsEnabled(fl EnabledFeatures) bool {
+	return true
+}
+
+type SupersedeTask struct {
+	task *TaskRun
+}
+
+func (feature *SupersedeFeature) NewTaskFeature(task *TaskRun) TaskFeature {
+	task.featureArtifacts[supersededByName] = "superseded feature"
+	return &SupersedeTask{
+		task: task,
+	}
+}
+
+func (l *SupersedeTask) RequiredScopes() scopes.Required {
+	// let's not require any scopes, as I see no reason to control access to this feature
+	return scopes.Required{}
+}
+
+func (l *SupersedeTask) Start() *CommandExecutionError {
+	supersederURL := l.task.Payload.SupersederURL
+	if supersederURL == "" {
+		return nil
+	}
+	resp, _, err := httpbackoff.Get(supersederURL)
+	if err != nil {
+		return MalformedPayloadError(err)
+	}
+	decoder := json.NewDecoder(resp.Body)
+	var supersedes SupersedesServiceResponse
+	err = decoder.Decode(&supersedes)
+	if err != nil {
+		return MalformedPayloadError(err)
+	}
+	taskIDs := supersedes.TaskIDs
+	if l.task.TaskID != taskIDs[0] {
+		supersededByFile := filepath.Join(taskContext.TaskDir, supersededByPath)
+		err = gwconfig.WriteToFileAsJSON(
```

Comment:
It just happened to be a utility method that was in the gwconfig package. I could create a separate fileutil package, I guess. Is it ok to leave that for now, as non-critical, and we change that later? Nice spot.